### PR TITLE
Sga 27 implement fish purchase functionality

### PR DESCRIPTION
**Pull request by paulius-mongirdas**
- Marketplace no longer uses mock item data. Now it reads/writes all data from/to database (including item images).
- Downloaded images are stored in memoryCache and diskCache. This allows for faster image loading times.
- Fully implemented buy/sell operations in marketplace.
- Added conditions for buy/sell operations: users cannot own more than one copy of an item and users can only sell the item if they already own it.
Screenshot_20240412_001001

### CODE REVIEW

File: app/src/main/java/com/github/k409/fitflow/ui/FitFlowApp.kt [hunk 1]
```diff
@@ -98,6 +102,25 @@ fun FitFlowApp(
 
     val snackbarHostState = remember { SnackbarHostState() }
 
+    val context = LocalContext.current
+    // ImageLoader has its own memory cache, disk cache, and OkHttpClient so it should be initialized only once in app
+    ImageLoader.Builder(context)
```
Comment (mantasjasikenas) [inline app/src/main/java/com/github/k409/fitflow/ui/FitFlowApp.kt]: Move this to FitFlowApplication. In Application class, implement the ImageLoaderFactory interface and override the newImageLoader() function. In method should be this logic.

File: app/src/main/java/com/github/k409/fitflow/ui/common/item/ItemCard.kt [hunk 2]
```diff
@@ -17,26 +17,48 @@ import androidx.compose.material3.CardDefaults
...
 import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import coil.compose.AsyncImage
+import coil.decode.SvgDecoder
+import coil.request.ImageRequest
+import com.github.k409.fitflow.R
+import com.github.k409.fitflow.ui.screen.market.MarketViewModel
+import kotlinx.coroutines.launch
 
+@SuppressLint("CoroutineCreationDuringComposition")
```
Comment (mantasjasikenas) [inline app/src/main/java/com/github/k409/fitflow/ui/common/item/ItemCard.kt]: Remove this

File: app/src/main/java/com/github/k409/fitflow/ui/common/item/ItemCard.kt [hunk 3]
```diff
@@ -17,26 +17,48 @@ import androidx.compose.material3.CardDefaults
...
     removeButtonText: String,
     onRemoveClick: () -> Unit = {},
+    removeButtonEnabled: Boolean,
     coinIcon: @Composable () -> Unit,
+    marketViewModel: MarketViewModel = hiltViewModel(),
 ) {
     val colors = MaterialTheme.colorScheme
+    val coroutineScope = rememberCoroutineScope()
+    var imageDownloadUrl by remember { mutableStateOf("") }
+    coroutineScope.launch {
```
Comment (mantasjasikenas) [inline app/src/main/java/com/github/k409/fitflow/ui/common/item/ItemCard.kt]: Wrap this in LaunchedEffect

File: app/src/main/java/com/github/k409/fitflow/ui/screen/market/MarketScreen.kt [hunk 4]
```diff
@@ -20,30 +21,44 @@ import androidx.compose.ui.Modifier
...
 import com.github.k409.fitflow.R
 import com.github.k409.fitflow.ui.common.ConfirmDialog
+import com.github.k409.fitflow.ui.common.FitFlowCircularProgressIndicator
 import com.github.k409.fitflow.ui.common.item.CategorySelectHeader
 import com.github.k409.fitflow.ui.common.item.InventoryItemCard
-import com.github.k409.fitflow.ui.common.item.Item
-import com.github.k409.fitflow.ui.common.item.mockDecorations
-import com.github.k409.fitflow.ui.common.item.mockFishes
 
+@SuppressLint("CoroutineCreationDuringComposition")
```
Comment (mantasjasikenas) [inline app/src/main/java/com/github/k409/fitflow/ui/screen/market/MarketScreen.kt]: Remove this

**Review by mantasjasikenas — CHANGES_REQUESTED** [review]
Please move all texts to string recourses. This will help when we will be adding other languages support.

**Review by mantasjasikenas — APPROVED** [review]

**Review by ugnspa — APPROVED** [review]

**Pull request merged**